sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs

```diff
@@ -3383,6 +3383,89 @@ public async Task ReadAsync()
             TestHelper.AssertSequenceEqual(data, actual.ToArray());
         }
 
+        [RecordedTest]
+        [ServiceVersion(Min = DataLakeClientOptions.ServiceVersion.V2024_05_04)]
+        public async Task ReadAsyncACL()
+        {
+            await using DisposingFileSystem test = await GetNewFileSystem(publicAccessType: PublicAccessType.None);
+            DataLakeDirectoryClient directory = await test.FileSystem.CreateDirectoryAsync(GetNewDirectoryName());
+
+            DataLakeFileClient fileClient = InstrumentClient(directory.GetFileClient(GetNewFileName()));
+
+            DataLakePathCreateOptions options = new DataLakePathCreateOptions
+            {
+                AccessOptions = new DataLakeAccessOptions
+                {
+                    AccessControlList = AccessControlList
+                }
+            };
+
+            await fileClient.CreateAsync(options: options);
+
+            // Arrange
+            var data = GetRandomBuffer(Constants.KB);
+            using (var stream = new MemoryStream(data))
+            {
+                await fileClient.AppendAsync(stream, 0);
```

Comment:
Nit: we don't actually need to append or flush data to this file for this test.  We also only need to check the ACLs in the assert.

Reply:
I know that we don't necessarily need to append and flush, I just felt having some data as a sanity check to read to make sure that ReadAsync method still correctly works under the conditions of appending and flushing data + acls, so I'll just keep it, feel free to reopen this if you think it needs to be removed